### DISCUSSION TITLE
flask项目无法实现检测文件变动时自动重启服务器

### DISCUSSION THREAD

**Discussion by kurt-steiner** [greyli/helloflask#236] [Q&A]
我想用到热部署，按照网上的教程 
```python3
if __name__ == '__main__':
    app.run(debug=True)
```  
但是这没有什么用

**Comment by greyli** [comment]
你的意思是让程序代码发生变动的时候自动重启开发服务器？你是怎么启动程序的？

**Comment by kurt-steiner** [comment]
flask run --reload

**Comment by greyli** [comment]
试试通过设置 FLASK_ENV 开启调试模式：
```
$ export FLASK_ENV=development  # Windows 用 set FLASK_ENV=development
$ flask run
```

**Comment by kurt-steiner** [comment]
!!! 这么长时间我都快忘了，麻烦您再帮我看看 

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/46296608/93705014-38f6b500-fb4c-11ea-8982-485d1cad42ee.gif)
![ezgif com-video-to-gif (1)](https://user-images.githubusercontent.com/46296608/93705049-8e32c680-fb4c-11ea-88ef-768d13c8af58.gif)  
有时候能检测到文件改动，一会就没用了，会不会我的操作出问题了？？
我的平台是**Ubuntu**, 创建虚拟环境
```bash
python3 -m venv env
. env/bin/activate
```

**Comment by greyli** [comment]
估计是 reloader 在特定操作系统的 bug，可以试下安装 watchdog：

```
$ pip install watchdog
```

安装后 Flask 会用它来检测文件变动（如果已经安装那就卸载掉试试……）。

（另，开启调试模式后不需要再传入 --reload 参数）

**Comment by kurt-steiner** [comment]
还是没什么用啊，可能是我的系统有问题，顺便问一下，搭建这个虚拟环境是必须的吗

**Comment by greyli** [comment]
或许可以换个编辑器试一下，记得要手动保存改动或是开启自动保存。

> 搭建这个虚拟环境是必须的吗

并不是必须的，主要的用处是方便记录当前项目的依赖列表。

**Comment by kurt-steiner** [comment]
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/46296608/95828225-4f68e880-0d67-11eb-91f4-178c88d7c9df.gif)
![flask1](https://user-images.githubusercontent.com/46296608/95828240-53950600-0d67-11eb-8bb6-c44420771aee.gif)

再次打扰，补充下完整的gif
